### ISSUE TITLE
Export RT-DETR to onnx get wrong parameters

### ISSUE THREAD

**Issue by civat** [ultralytics/ultralytics#19753]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I used the following code to export pre-trained RT-DETR model to onnx:

```
from ultralytics import RTDETR

model = RTDETR("detection/ultralytics/runs/rtdert_ad/rtdert_resnet50_tiny_x_e3003/weights/best.pt")
model.export(
    format="onnx", 
    imgsz=(960, 480), 
    half=False, 
    simplify=False, 
    opset=None,
    device="0", 
    dynamic=False)
```

But the output onnx model has different parameters with the RTDETR model.
For example, for the first 7*7 Conv, the  onnx model's metrics (in NETRON) are (too large):
```
min: -2.9261393547058105
max: 3.2974531650543213
```

while the RTDETR first 7*7 Conv's metrics are:
```
min: -0.305908203125
max: 0.3447265625
```

Here is the version of installed packages:
```
onnx                     1.16.1
onnxruntime              1.20.1
onnxruntime-gpu          1.20.0
onnxsim                  0.4.36
onnxslim                 0.1.48
torch                    2.4.1+cu118
ultralytics              8.3.73
ultralytics-thop         2.0.14
```

Any idea to deal with this problem? 
Thanks a lot.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @civat, thank you for your detailed report and interest in Ultralytics 🚀!  

We recommend reviewing the [Ultralytics Docs](https://docs.ultralytics.com/) where you can find [exporting](https://docs.ultralytics.com/modes/export/) guidelines and examples for various formats, including ONNX. This might help clarify any potential issues during the export process.

If this is a 🐛 Bug Report, we kindly request a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it effectively. Your provided code is helpful, but additional details such as any modifications to the model or dataset specifics would assist further.

In the meantime:  
1. Ensure you're using the latest `ultralytics` package and dependencies. Upgrade by running:
```bash
pip install -U ultralytics
```
2. Double-check that your environment matches the required versions of [PyTorch](https://pytorch.org/), ONNX, and related libraries.  
3. If you're encountering discrepancies in parameter values, consider testing with different export options (e.g., using `dynamic=True` or specifying an `opset`) as these can influence exported model behavior.  

For community support, you can join our [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧, visit [Discourse](https://community.ultralytics.com/) for in-depth discussions, or connect with fellow users on our [Subreddit](https://reddit.com/r/Ultralytics).

## Verified Environments

YOLO is tested in various environments with dependencies preinstalled. You can try running your export in one of these environments for consistent results:
- **Notebooks** with free GPUs: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>  
- **Google Cloud** Deep Learning VM: Follow the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)  
- **Amazon** Deep Learning AMI: Follow the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)  
- **Docker Image**: Use the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>  
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are passing, verifying the correct operation of YOLO tasks and modes across platforms.

This is an automated response to assist you faster 😊. An Ultralytics engineer will review your issue and provide further guidance soon.

**Comment by Y-T-G** [comment]
Does it affect the detection?

**Comment by Y-T-G** [comment]
The change Conv parameters is simply due to BN fusion. It's normal. You will get the same value in PyTorch if you do `model.fuse()`.

**Comment by Y-T-G** [comment]
> > Does it affect the detection?
> 
> Yes. onnx model performs much worse than the RTDETR model.

It's exactly the same in my test.

```python
In [6]: model = RTDETR("rtdetr-l.onnx")

In [7]: results = model()
WARNING ⚠️ 'source' is missing. Using 'source=/ultralytics/ultralytics/assets'.
Loading rtdetr-l.onnx for ONNX Runtime inference...
Using ONNX Runtime CPUExecutionProvider

results[0image 1/2 /ultralytics/ultralytics/assets/bus.jpg: 640x640 4 persons, 1 bus, 1 traffic light, 1 fire hydrant, 2 ties, 6515.5ms
].boxes.confimage 2/2 /ultralytics/ultralytics/assets/zidane.jpg: 640x640 2 persons, 2 ties, 8294.4ms
Speed: 3.2ms preprocess, 7405.0ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 640)

In [8]: results[0].boxes.conf
Out[8]: tensor([0.9487, 0.9553, 0.9258, 0.9485, 0.8623, 0.2834, 0.3393, 0.2660, 0.2611])

In [9]: model = RTDETR("rtdetr-l.pt")

In [10]: results = model()
WARNING ⚠️ 'source' is missing. Using 'source=/ultralytics/ultralytics/assets'.

image 1/2 /ultralytics/ultralytics/assets/bus.jpg: 640x640 4 persons, 1 bus, 1 traffic light, 1 fire hydrant, 2 ties, 317.5ms
image 2/2 /ultralytics/ultralytics/assets/zidane.jpg: 640x640 2 persons, 2 ties, 321.9ms
Speed: 2.4ms preprocess, 319.7ms inference, 0.4ms postprocess per image at shape (1, 3, 640, 640)

In [11]: results[0].boxes.conf
Out[11]: tensor([0.9487, 0.9553, 0.9258, 0.9485, 0.8623, 0.2834, 0.3393, 0.2660, 0.2611])
```

**Comment by civat** [comment]
> The change in Conv parameters is simply due to BN fusion. It's normal. You will get the same value in PyTorch if you do `model.fuse()`.



> The change in Conv parameters is simply due to BN fusion. It's normal. You will get the same value in PyTorch if you do `model.fuse()`.

I got it. It's exactly the same for both. Thank you.
Now my concern is that: how to preprocess the input image when using onnxruntime? 
Here is my preprocessing:

```
def preprocess_image(image_path, input_size=(480, 960)):
    image = cv2.imread(image_path)     # [H, W, C]
    height, width, _ = image.shape
    image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
    image = cv2.resize(image, input_size) 
    image = image.astype(np.float32) / 225

    # from [W, H, C] to [C, W, H]
    image = np.transpose(image, (2, 0, 1))
    # add batch dimension: [B, C, W, H]
    image = np.expand_dims(image, axis=0)
    return image, height, width
```

**Comment by Y-T-G** [comment]
You need to divide by 255, not 225
